apache/superset
# feat(helm): Support for flower and websocket containers

### SUMMARY

This PR brings a number of updates to the Helm chart:

- Support running an optional `flower` ~~sidecar container in the celery beat pod~~ deployment, to visualize the tasks (but NOT exposed through the ingress, since most users probably don't want to expose this publicly - either a new ingress needs to be deployed, or a port-forward to the flower service is needed)
- Support the optional deployment of a `superset-websocket` server in a separate pod, together with associated services and ingress path within the same host(s)
- Reworked the probes to make them fully customizable through the `values.yaml`, added `startupProbes`, as well as a `livenessProbe` for the celery workers (that runs a `celery inspect ping` command)
- Added an auto-generated `README`, using [helm-docs](https://github.com/norwoodj/helm-docs), to document the contents of `values.yaml` (and should be visible from artifacthub)
- Some minor cleanups, in particular switched all the `indent` to `nindent` so that we can apply indentation and improve readability

The more controversial change is also that I've removed the schema definition... I'm sorry but I've really find it too hard to work with, and was making iterative developments of the chart quite painfully slow, while not necessarily helping catch too many issues...

I guess this is debatable, and if that's a problem I can work on restoring it, but I personally suggest to drop it since I think it creates unnecessary work - unless there's a tool to generate it that I'm not aware of?

### TESTING INSTRUCTIONS

- Deploy a new release of the chart. To enable the new features:
  - set `supersetCeleryBeatFlower.enabled=true` to see the flower sidecar container in the celerybeat pod, and access the UI by port-forwarding its port 5555. WARNING this requires a custom Superset image that has `flower<1.0.0` installed (recent versions are causing a dependency conflict)
  - set `supersetWebsockets.enabled=true` to see the superset-websocket pod (by default a custom image I've built from our fork, since there's no official one). It assumes that both pods are exposed through an ingress on the same domain (the built-in ingress will do this automatically) and requires various things:
    - `JWT_SECRET` set as a secret env variable - it will be passed to both Superset and websocket servers
    - in `superset_config.py`:
      - `GLOBAL_ASYNC_QUERIES_TRANSPORT=ws`
      - A proper value for `GLOBAL_ASYNC_QUERIES_WEBSOCKET_URL` (I use `f"{re.sub('^http','ws',env('BASE_URL'))}/ws"`)
    - `GLOBAL_ASYNC_QUERIES_JWT_SECRET=env("JWT_SECRET")`

This was tested successfully on Superset 1.5.2

### ADDITIONAL INFORMATION
